src/server/player_sao.h

```diff
@@ -177,12 +199,32 @@ class PlayerSAO : public UnitSAO
 	v3f getEyeOffset() const;
 	float getZoomFOV() const;
 
+	const std::unordered_map<std::string, PhysicsModifier> &getPhysicsModifiers()
+	{
+		return m_physics_modifiers;
+	}
+
+	void setPhysicsModifier(const std::string &key, const PhysicsModifier &modifier);
+	void deletePhysicsModifier(const std::string &key);
+
+	// Not const - physics modifier is cached
+	const PhysicsModifier &getEffectivePhysics();
+
 	inline Metadata &getMeta() { return m_meta; }
 
 private:
 	std::string getPropertyPacket();
 	void unlinkPlayerSessionAndSave();
-	std::string generateUpdatePhysicsOverrideCommand() const;
+	std::string generateUpdatePhysicsOverrideCommand();
+
+	// Not const since it may cached
```

Comment:
ah makes sense
C++ has [`mutable`](https://en.cppreference.com/w/cpp/language/cv#mutable_specifier) for these situations